h2o-py/tests/testdir_algos/gbm/pyunit_PUBDEV_7578_gbm_reproducibility.py

```diff
@@ -0,0 +1,240 @@
+    else:
+        if throwError:
+            assert False, "The two arrays are of different size!"
+        else:
+            return False
+
+def extract_from_twoDimTable(metricOfInterest, fieldOfInterest, takeFirst=False):
+    """
+    Given a fieldOfInterest that are found in the model scoring history, this function will extract the list
+    of field values for you from the model.
+
+    :param aModel: H2O model where you want to extract a list of fields from the scoring history
+    :param fieldOfInterest: string representing a field of interest.
+    :return: List of field values or None if it cannot be found
+    """
+
+    allFields = metricOfInterest._col_header
+    if fieldOfInterest in allFields:
+        cellValues = []
+        fieldIndex = allFields.index(fieldOfInterest)
+        for eachCell in metricOfInterest.cell_values:
+            cellValues.append(eachCell[fieldIndex])
+            if takeFirst:   # only grab the result from the first iteration.
+                break
+        return cellValues
+    else:
+        return None
+
+def extract_scoring_history_field(aModel, fieldOfInterest, takeFirst=False):
+    """
+    Given a fieldOfInterest that are found in the model scoring history, this function will extract the list
+    of field values for you from the model.
+
+    :param aModel: H2O model where you want to extract a list of fields from the scoring history
+    :param fieldOfInterest: string representing a field of interest.
+    :return: List of field values or None if it cannot be found
+    """
+    return extract_from_twoDimTable(aModel._model_json["output"]["scoring_history"], fieldOfInterest, takeFirst)
+
+
+def model_seed_sorted(model_list):
+    """
+    This function is written to find the seed used by each model in the order of when the model was built.  The
+    oldest model metric will be the first element.
+    :param model_list: list of models built sequentially that contains metric of interest among other fields
+    :return: model seed sorted by order of building
+    """
+
+    model_num = len(model_list)
+
+    model_seed_list = [None] * model_num
+
+
+    for index in range(model_num):
+        for pIndex in range(len(model_list.models[0]._model_json["parameters"])):
+            if model_list.models[index]._model_json["parameters"][pIndex]["name"]=="seed":
+                model_seed_list[index]=model_list.models[index]._model_json["parameters"][pIndex]["actual_value"]
+                break
+    print("model seed list before sort: ")
+    model_seed_list_string = ','.join(str(x) for x in model_seed_list[0:model_num])
+    print(model_seed_list_string)
+    model_seed_list.sort()
+    print("model seed list after sort: ")
+    model_seed_list_string = ','.join(str(x) for x in model_seed_list[0:model_num])
+    print(model_seed_list_string)
+    return model_seed_list
+
+
+def standalone_test(test):
```

Comment:
duplicates pyunit_utils, why?

Reply:
Sorry, I just use this attached test from the JIRA ticket without paying attention to the whole code... Thank you for your suggestion. I will change it.